app/scripts/controllers/transactionsBodyModal.js

```diff
@@ -1,16 +1,49 @@
-import { beautifyIndent, returnContentType } from '../utils'
+import { beautifyIndent, returnContentType, retrieveBodyProperties } from '../utils'
+import { Api } from '../services/rest'
 
-export function TransactionsBodyModalCtrl ($scope, $uibModalInstance, bodyData) {
+export function TransactionsBodyModalCtrl ($scope, $uibModalInstance, config, Alerting, bodyData) {
   $scope.bodyData = bodyData
+  const defaultLengthOfBodyToDisplay = config.defaultLengthOfBodyToDisplay
 
+  $scope.partialBody = false
   // transform body with indentation/formatting
-  if ($scope.bodyData.content) {
+  if ($scope.bodyData && $scope.bodyData.content) {
     if (bodyData.headers && returnContentType(bodyData.headers)) {
       const bodyTransform = beautifyIndent(returnContentType(bodyData.headers), bodyData.content)
       $scope.bodyData.content = bodyTransform.content
     }
   }
 
+  if (
+    $scope.bodyData &&
+    $scope.bodyData.transactionId &&
+    $scope.bodyData.bodyId
+  ) {
+    $scope.retrieveBodyData = function (start=0, end=defaultLengthOfBodyToDisplay) {
+      Api.TransactionBodies($scope.bodyData.transactionId, $scope.bodyData.bodyId, start, end).then(response => {
+        const { start, end, bodyLength } = retrieveBodyProperties(response)
+
+        if (start && end && bodyLength) {
+          $scope.bodyStart = start
+          $scope.bodyEnd = end
+          $scope.bodyLength = bodyLength
+
+          if ((bodyLength - end) > 1) {
+            $scope.partialBody = true
+          }
+        }
+
+        if (bodyData.headers && returnContentType(bodyData.headers)) {
+          let bodyTransform = beautifyIndent(returnContentType(bodyData.headers), response.data)
+          $scope.bodyData.content = bodyTransform.content
+        }
+      }).catch(err => {
+        Alerting.AlertAddServerMsg(err.status)
+      })
+    }
+    $scope.retrieveBodyData()
```

Comment:
Since we already have the body data we don't need to call this here. It can be called later when we need to laod more of the body.